core/src/main/java/kafka/log/remote/RemoteLogManager.java

```diff
@@ -1680,9 +1671,9 @@ public FetchDataInfo read(RemoteStorageFetchInfo remoteStorageFetchInfo) throws
         OptionalInt epoch = OptionalInt.empty();
 
         if (logOptional.isPresent()) {
-            Option<LeaderEpochFileCache> leaderEpochCache = logOptional.get().leaderEpochCache();
-            if (leaderEpochCache != null && leaderEpochCache.isDefined()) {
```

Comment:
@junrao It's odd that we were checking for `null` here - do you know if we can remove this?

Reply:
Yes, we can remove this check.